app/lib/common/widget/ChatOverview.dart

```diff
@@ -68,24 +68,12 @@ class ChatListItem extends StatelessWidget {
           ),
         );
       },
-      leading: FutureBuilder<Uint8List>(
-        future: room.avatar().then((fb) => fb.asTypedList()),
-        // a previously-obtained Future<String> or null
-        builder: (BuildContext context, AsyncSnapshot<List<int>> snapshot) {
-          if (snapshot.hasData) {
-            return CircleAvatar(
-              radius: 25,
-              backgroundImage:
-                  MemoryImage(Uint8List.fromList(snapshot.requireData)),
-            );
-          } else {
-            return CircleAvatar(
-              radius: 25,
-              backgroundColor: Colors.grey[700],
-              child: SvgPicture.asset('assets/images/people.svg'),
-            );
-          }
-        },
+      leading: CustomAvatar(
+        avatar: room.avatar().then((fb) => fb.asTypedList()),
```

Comment:
isn't this kind of always the pattern? Shouldn't we then put the `then( (fb) ..` part into the component rather than repeating it every time?